src/Platforms/OraclePlatform.php

```diff
@@ -59,11 +59,11 @@ public static function assertValidIdentifier($identifier)
      */
     public function getSubstringExpression($string, $start, $length = null)
     {
-        if ($length !== null) {
-            return sprintf('SUBSTR(%s, %d, %d)', $string, $start, $length);
+        if ($length === null) {
+            return 'SUBSTR(' . $string . ', ' . $start . ')';
         }
 
-        return sprintf('SUBSTR(%s, %d)', $string, $start);
+        return 'SUBSTR(' . $string . ', ' . $start . ', ' . $length . ')';
```

Comment:
I think I would stick with `sprintf` here, and just switch the `%d` to `%s`, that would work, wouldn't it?

Reply:
The new code is based on the implementation I found in **AbstractPlatform.php**, which doesn't use `sprintf`.
https://github.com/doctrine/dbal/blob/7ef157430c43e41345205ee934f0409854f018bd/src/Platforms/AbstractPlatform.php#L1120-L1127
I think it would be more consistent to do it that way when you look at the code base it its entirety. These are the other implementations of `getSubstringExpression` I found. None is using `sprintf`.
https://github.com/doctrine/dbal/blob/85b98cb23c8af471a67abfe14485da696bcabc2e/src/Platforms/SQLServerPlatform.php#L1214-L1221
https://github.com/doctrine/dbal/blob/85b98cb23c8af471a67abfe14485da696bcabc2e/src/Platforms/SqlitePlatform.php#L120-L127
https://github.com/doctrine/dbal/blob/85b98cb23c8af471a67abfe14485da696bcabc2e/src/Platforms/DB2Platform.php#L902-L909
https://github.com/doctrine/dbal/blob/7ef157430c43e41345205ee934f0409854f018bd/src/Platforms/PostgreSQLPlatform.php#L81-L88

If you want to keep the commit small however, we can stick to `sprintf`.

Reply:
Good point, please ignore me on that one.